src/DatabaseLogReader.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace SESP;
+
+use DatabaseLogEntry;
+use DatabaseBase;
+use MWTimestamp;
+use Title;
+use User;
+
+class DatabaseLogReader {
+
+	// The parameters for our query.
+	private $query;
+
+	// The current log
+	private $log;
+
+	// Don't run multiple queries if we don't have to
```

Comment:
This comment is misplaced. Comments for fields are supposed to describe them. I'd just add a type indication here. And move the static field to the top (or bottom if you prefer) of the list, not in between instance fields.

Reply:
And of course using static fields and other forms of global state is a big anti-pattern. Better to inject an actual cache service.